internal/storage/image.go

```diff
@@ -94,6 +94,7 @@ type imageService struct {
 	ctx                  context.Context
 	config               *config.Config
 	regexForPinnedImages []*regexp.Regexp
+	imageRemovalLock     sync.Map
```

Comment:
Question about locks here: we already have `imageCacheLock`. Which is also just a normal `sync.Mutex`, nothing as fancy as `sync.Map`. Would it be possible to use it too? Rename to some generic `imageLock` or just embedd directly in the type, and then use? Or do we need `sync.Map`?

Reply:
Looking at things more closely.  I suppose, having a per-image lock has the advantage of removing unnecessary contention since it allows for more granural locking.